src/main/scala/org/constellation/domain/healthcheck/HealthCheckConsensus.scala

```diff
@@ -0,0 +1,569 @@
+                    apiClient.healthcheck.fetchPeerHealthStatus(FetchPeerHealthStatus(roundIds.toSortedSet, id.some)),
+                    unboundedBlocker
+                  )(proxyPeer.peerMetadata.toPeerClientMetadata.copy(port = "9003"))
+                } yield response
+              case Right(peerData) =>
+                PeerResponse.run(
+                  apiClient.healthcheck.fetchPeerHealthStatus(FetchPeerHealthStatus(roundIds.toSortedSet)),
+                  unboundedBlocker
+                )(peerData.peerMetadata.toPeerClientMetadata.copy(port = "9003"))
+            }
+          }.handleErrorWith(
+            _ => FetchingFailed(id).asInstanceOf[FetchProposalError].asLeft[SendPerceivedHealthStatus].pure[F]
+          )
+        }.map(_.bimap(id -> _, id -> _))
+      }
+      (failed, succeeded) = responses.separate
+      _ <- succeeded.traverse { case (_, proposal) => processProposal(proposal.consensusHealthStatus) }
+      _ <- failed.traverse {
+        case (id, error) =>
+          logger.debug(error)(s"Error during fetching healtcheck proposal from peer with id=$id.")
+      }
+    } yield ()
+
+  private def randomlyPickProxyPeer(proxyPeers: NonEmptySet[PeerData]): F[PeerData] = F.delay(
+    Random.shuffle(proxyPeers.toList).head
+  )
+
+  private def sendProposal(): F[Unit] =
+    for {
+      isAcquired <- sendingProposalLock.tryAcquire
+      _ <- if (isAcquired)
+        sendHealthStatus()
+          .handleErrorWith(
+            e =>
+              sendingProposalLock.release >> logger
+                .error(e)("Error sending healtcheck proposal") /*>> e.raiseError[F, Unit]*/
+          )
+      else
+        F.unit
+      _ <- sendingProposalLock.release
+    } yield ()
+
+  private def sendHealthStatus(): F[Unit] =
+    for {
+      peers <- roundPeers.get
+        .map(_.filterNot { case (_, rd) => rd.receivedProposal })
+      healthStatus <- ownConsensusHealthStatus.get
+      responses <- peers.toList.traverse {
+        case (id, roundData) => {
+          {
+            roundData.peerData match {
+              case Left(proxyPeers) =>
+                for {
+                  proxyPeer <- randomlyPickProxyPeer(proxyPeers.peersData)
+                  response <- PeerResponse.run(
+                    apiClient.healthcheck.sendPeerHealthStatus(SendPerceivedHealthStatus(healthStatus, id.some)),
+                    unboundedBlocker
+                  )(proxyPeer.peerMetadata.toPeerClientMetadata.copy(port = "9003"))
+                } yield response
+              case Right(peerData) =>
+                PeerResponse.run(
+                  apiClient.healthcheck.sendPeerHealthStatus(SendPerceivedHealthStatus(healthStatus)),
+                  unboundedBlocker
+                )(peerData.peerMetadata.toPeerClientMetadata.copy(port = "9003"))
+            }
+          }.handleErrorWith(_ => SendingFailed(id).asInstanceOf[SendProposalError].asLeft[Unit].pure[F])
```

Comment:
if you want to provide a value, not an IO[value], you can use `handleError` instead of `handleErrorWith`